snomed/com.b2international.snowowl.snomed.datastore.server/src/com/b2international/snowowl/datastore/server/snomed/version/SnomedPublishManager.java

```diff
@@ -234,15 +242,43 @@ private boolean isCoreComponent(final EClass eClass) {
 
 	/**Updates all new module dependency reference set members.
 	 * @param date */
-	private void adjustNewModuleDependencyRefSetMembers(Date effectiveTime) {
-		final SnomedRegularRefSet moduleDependencyRefSet = getEditingContext().lookup(REFSET_MODULE_DEPENDENCY_TYPE, SnomedRegularRefSet.class);
-		for (final SnomedModuleDependencyRefSetMember member : newModuleDependencyRefSetMembers) {
-			adjustRelased(member);
-			adjustEffectiveTime(member, effectiveTime);
-			moduleDependencyRefSet.getMembers().add(member);
+	private void adjustDependencyRefSetMembers(Date effectiveTime) {
+		// Update existing, add new members to moduleDependecyRefSet
+		if (!changedModuleDependencyRefSetMembers.isEmpty()) {
+			final SnomedRegularRefSet moduleDependencyRefSet = getEditingContext().lookup(REFSET_MODULE_DEPENDENCY_TYPE, SnomedRegularRefSet.class);
+			changedModuleDependencyRefSetMembers.forEach(moduleToRefComp -> {
+				boolean modified = false;
+				for (SnomedRefSetMember refSetmember : moduleDependencyRefSet.getMembers()) {
```

Comment:
Almost good, but we need to prepare for certain extensions who manage their module dependency members in a different way than IHTSDO does. Let's talk about it when you are making changes the next time.